src/Form/Form.js

```diff
@@ -370,14 +370,22 @@ class Form {
 
     categorizeInputs () {
         const selector = this.matching.cssSelector('formInputsSelector')
+        // If there's no form container and it's just a lonely input field (this.form is an input field)
         if (this.form.matches(selector)) {
             this.addInput(this.form)
         } else {
-            let foundInputs = this.form.querySelectorAll(selector)
-            // If the markup is broken form.querySelectorAll may not return the fields, so we select from the parent
-            if (foundInputs.length === 0 && this.form instanceof HTMLFormElement && this.form.length > 0) {
-                foundInputs = this.form.parentElement?.querySelectorAll(selector) || foundInputs
+            /** @type {Element[] | NodeList} */
+            let foundInputs = []
+
+            if (this.form instanceof HTMLFormElement) {
+                // For form elements we use .elements to catch fields outside the form itself using the form attribute.
+                // It also catches all elements when the markup is broken.
+                // We use .filter to avoid fieldset, button, textarea etc.
+                foundInputs = [...this.form.elements].filter(el => el.matches(selector))
```

Comment:
The comment says it all, it's to ensure that fields outside the form tree, but with the `form` attribute are accounted for.